File: app/src/main/java/com/codemitry/scanme/history/HistoryAction.java
package com.codemitry.scanme.history;

import com.codemitry.scanme.R;
import com.codemitry.scanme.barcode.Barcode;

import java.io.Serializable;


public class HistoryAction implements Serializable {
    private Actions action;
    private Barcode barcode;

    public HistoryAction(Actions action, Barcode barcode) {
        this.action = action;
        this.barcode = barcode;
    }

    public HistoryAction(Actions action, com.google.mlkit.vision.barcode.Barcode barcode) {
        this(action, Barcode.getBarcode(barcode));
    }

    public enum Actions {
        SCAN, CREATE;

        public static int getString(Actions action) {
            switch (action) {
                case SCAN:
                    return R.string.qr_scanning;
                case CREATE:
                    return R.string.qr_creation;
                default:
                    return -1;
            }
        }
    }

    public Actions getAction() {
        return action;
    }

    public Barcode getBarcode() {
        return barcode;
    }

}

File: app/src/main/java/com/codemitry/scanme/history/HistoryAction.kt
package com.codemitry.scanme.history

import com.codemitry.qr_code_generator_lib.qrcode.Barcode
import com.codemitry.scanme.R
import com.codemitry.scanme.barcode.BarcodeAdapter
import java.io.Serializable

data class HistoryAction(val action: Actions, val barcode: Barcode) : Serializable {

    constructor(action: Actions, barcode: com.google.mlkit.vision.barcode.Barcode) : this(action, BarcodeAdapter.barcode((barcode)))


    enum class Actions {
        SCAN, CREATE;

        companion object {
            fun getString(action: Actions) =
                    when (action) {
                        SCAN -> R.string.qr_scanning
                        CREATE -> R.string.qr_creation
                    }
        }
    }

}